src/briefcase/platforms/macOS/__init__.py

```diff
@@ -241,21 +241,20 @@ def select_identity(self, identity=None):
                         f"Invalid code signing identity {identity!r}"
                     ) from e
 
-        if len(identities) == 0:
-            raise BriefcaseCommandError(
-                "No code signing identities are available: see "
-                "https://briefcase.readthedocs.io/en/stable/how-to/code-signing/macOS.html"
-            )
-        elif len(identities) == 1:
-            identity, identity_name = list(identities.items())[0]
-        else:
-            self.input.prompt()
-            self.input.prompt("Select code signing identity to use:")
-            self.input.prompt()
-            identity = select_option(identities, input=self.input)
-            identity_name = identities[identity]
-            self.logger.info(
-                f"""
+        identities["-"] = "ad-hoc"
+        self.input.prompt()
+        self.input.prompt(
+            'Select code signing identity to use. "ad-hoc" will result '
+            "in a package that can run locally, but cannot be "
+            "redistributed. If your identity is not listed, please see "
+            "https://briefcase.readthedocs.io/en/stable/how-to/code-signing/macOS.html\n"
+            "To skip this prompt in the future, use `--adhoc-sign` or `--identity=<identity>`:"
```

Comment:
The "use -i in future" advice isn't needed; it's provided after an identity is selected.